scout/adapter/mongo/case.py

```diff
@@ -237,6 +238,34 @@ def cases(self, owner=None, collaborator=None, query=None, skip_assigned=False,
                     {'individuals.display_name': {'$regex': name_query}},
                 ]
 
+        if within_days:
+            verbs = []
+
+            if has_causatives:
+                verbs.append('mark_causative')
+            if finished:
+                verbs.append('archive')
+                verbs.append('mark_causative')
+            if reruns:
+                verbs.append('rerun')
+            if research_requested:
+                verbs.append('open_research')
+
+            days_datetime = datetime.datetime.now() - datetime.timedelta(days=within_days)
+            # Look up 'mark_causative' events added since specified number days ago
+            event_query = {
+                'category': 'case',
+                'verb': {'$in': verbs},
+                'created_at': {'$gte': days_datetime}
+                }
+            recent_events = self.event_collection.find(event_query)
+            solved_cases = set()
```

Comment:
Maybe better to rename this to something else, since it's no more just the solved ones?